crates/matrix-sdk-indexeddb/src/crypto_store/migrations.rs

```diff
@@ -136,3 +196,226 @@ fn create_stores_for_v5(db: &IdbDatabase) -> Result<(), DomException> {
 
     Ok(())
 }
+
+fn create_stores_for_v6(db: &IdbDatabase) -> Result<(), DomException> {
+    // We want to change the shape of the inbound group sessions store. To do so, we
+    // first need to build a new store, then copy all the data over.
+    //
+    // But copying the data needs to happen outside the database upgrade process
+    // (because it needs async calls). So, here we create a new store for
+    // inbound group sessions. We don't populate it yet; that happens once we
+    // have done the upgrade to v6, in `migrate_data_for_v6`. Finally we drop the
+    // old store in create_stores_for_v7.
+
+    info!("Creating {}", keys::INBOUND_GROUP_SESSIONS);
+    let object_store = db.create_object_store(keys::INBOUND_GROUP_SESSIONS)?;
+
+    let mut params = IdbIndexParameters::new();
+    params.unique(false);
+    object_store.create_index_with_params(
+        keys::INBOUND_GROUP_SESSIONS_BACKUP_INDEX,
+        &IdbKeyPath::str("needs_backup"),
+        &params,
+    )?;
+
+    Ok(())
+}
+
+async fn migrate_data_for_v6(serializer: &IndexeddbSerializer, db: &IdbDatabase) -> Result<()> {
+    // The new store has been made for inbound group sessions; time to populate it.
+    let txn = db.transaction_on_multi_with_mode(
+        &[old_keys::INBOUND_GROUP_SESSIONS_OLD, keys::INBOUND_GROUP_SESSIONS],
+        IdbTransactionMode::Readwrite,
+    )?;
+
+    let old_store = txn.object_store(old_keys::INBOUND_GROUP_SESSIONS_OLD)?;
+    let new_store = txn.object_store(keys::INBOUND_GROUP_SESSIONS)?;
+
+    let row_count = old_store.count()?.await?;
+    info!(row_count, "Migrating inbound group session data");
```

Comment:
With the above suggestion for the const names:

```suggestion
    info!(row_count, "Migrating inbound group session v1 data to v2");
```